src/Uno.UI.Runtime.Skia.Wpf/Input/WpfCorePointerInputSource.cs

```diff
@@ -230,21 +259,86 @@ private IntPtr OnWmMessage(IntPtr hwnd, int msg, IntPtr wparamOriginal, IntPtr l
 	#endregion
 
 	#region Convert helpers
-	private PointerEventArgs BuildPointerArgs(WpfMouseEventArgs args)
+	private PointerEventArgs BuildPointerArgs(InputEventArgs args)
 	{
 		if (args is null)
 		{
 			throw new ArgumentNullException(nameof(args));
 		}
 
-		var position = args.GetPosition(_hostControl);
-		var properties = BuildPointerProperties(args).SetUpdateKindFromPrevious(_previous?.CurrentPoint.Properties);
+		Point position;
+		PointerPointProperties properties;
+
+		uint pointerId;
+		if (args is WpfMouseEventArgs mouseEventArgs)
+		{
+			pointerId = 1;
+			position = mouseEventArgs.GetPosition(_hostControl);
+			properties = new()
+			{
+				IsLeftButtonPressed = mouseEventArgs.LeftButton == MouseButtonState.Pressed,
+				IsMiddleButtonPressed = mouseEventArgs.MiddleButton == MouseButtonState.Pressed,
+				IsRightButtonPressed = mouseEventArgs.RightButton == MouseButtonState.Pressed,
+				IsXButton1Pressed = mouseEventArgs.XButton1 == MouseButtonState.Pressed,
+				IsXButton2Pressed = mouseEventArgs.XButton2 == MouseButtonState.Pressed,
+				IsPrimary = true,
+				IsInRange = true
+			};
+		}
+		else if (args is StylusEventArgs stylusEventArgs)
+		{
+			pointerId = (uint)stylusEventArgs.StylusDevice.Id;
+			position = stylusEventArgs.GetPosition(_hostControl);
+
+			properties = new()
+			{
+				IsLeftButtonPressed = true,
```

Comment:
For touch, this has to be `false` for release and cancel.
For pen, it has to be `true` only when `!stylusEventArgs.InAir`.

Also for the pen do we have information about the barel button and the eraser (for the `IsRightButtonPressed` and `IsEraser`)?

Reply:
@dr1rrb Yeah. Get the barel button by this code:

```
        private void MainWindow_StylusMove(object sender, StylusEventArgs e)
        {
            if (e.StylusDevice.StylusButtons.GetStylusButtonByGuid(StylusPointProperties.BarrelButton.Id).StylusButtonState == StylusButtonState.Down)
            {
                
            }
        }
```

Reply:
Thank you @dr1rrb  and I create the PR in https://github.com/unoplatform/uno/pull/16517 . Could you review my code?